calico_mesos/calico_mesos.py

```diff
@@ -71,6 +72,19 @@ def calico_mesos():
     except KeyError:
         raise IsolatorException(ERROR_MISSING_ARGS)
 
+    # Unfortunately, netmodules passes in labels poorly, so move them
+    # around to match the agreed upon design.
+    if args.has_key('labels'):
+        # copy labels so we're not modifiying the original as we iterate
+        _labels = copy.deepcopy(args['labels'])
+        args['labels'] = {}
+        for label in _labels:
+            # Move the key and value into the labels dictionary
+            key = label['key']
+            val = label['value']
+            args['labels'][key] = val
+        _log.info("Fixed request to be: %s" % str(args))
```

Comment:
In fact, you could always populate labels so you don't have to check elsewhere by changing this to:

```
args['labels'] = {label['key']: label['value'] for label in args.get('labels', [])}
```

That way you can get rid of line 77 as well.